provd/internal/services/user/user_test.go

```diff
@@ -52,81 +152,55 @@ func TestCreateUser(t *testing.T) {
 		autoLogin bool
 		isAdmin   bool
 
-		accountsError bool
-		hostnameError bool
-
 		wantErr bool
-
-		accountsEndpoint string
-		hostnameEndpoint string
 	}{
 		"Successfully creates a user": {
-			realName:  "Ubuntu",
-			username:  "ubuntu",
-			password:  "password",
-			hostname:  "ubuntu",
+			realName:  "ok",
```

Comment:
So, as you can see, most of the tests have:

`
realName:  "ok",
username:  "ok",
password:  "ok",
hostname:  "ok",
```

which is tedious :)

I suggest that we use default values, don’t set them, then in the subtests:

```
if tc.username == "" {
   tc.username = "ok` // (I think a better name can be used too)
}
```

Then, you are going to tell me "yes, but what if I want the default value for that element (an empty string in that case", and this is where I answer "no problem, just have a convention". What we do in enterprise desktop/WSL is that we picked "-". That means "empty" (which is generally just one test case).

You then end up with:
```
if tc.username == "" {
   tc.username = "ok` // (I think a better name can be used too)
} else if tc.username == "-" {
   tc.username = ""
}
```

And if you want to be fancy and add a third one, you can even use a switch!

That way, you are going to remove many lines in the test case definition, and you can then ordering them to be more compact (as you have fewer fields, only you can have everything in one line for most of there), and as you are going to deviate from the default like "when hostname is empty", you set this hostname to empty as the **first** field to assign in your named struct for that test. That way, the test case become:

`"Error when hostname is empty": { hostname:  "ubuntu", wantErr:   true}`

as everything else is not relevant and then covered by our default set in the subtests.

Then, you can organize your tests cases with:

```
// Positive tests, the easiest first
< line break>
// Positive tests, but special cases or special behaviours, you can even have many sections delimited by lines break with a comment to illustrate groups of related special cases
< line break>
// Error cases
```

Some example of this:

You can see the struct definition is separated (as you do!): https://github.com/ubuntu/authd/blob/main/internal/services/pam/pam_test.go#L248

The tests cases are then grouped and most of time, one liner, very easy to read and see only the changed parameters impacted this use case: https://github.com/ubuntu/authd/blob/main/internal/services/pam/pam_test.go#L260

We do have some defaults for some of those values to avoid repetition: https://github.com/ubuntu/authd/blob/main/internal/services/pam/pam_test.go#L285

Oh, and while I am at it, nitpick, but `username, hostname, but realName` -> I think we can drop the capital N then :)